src/controllers/redeploy.go

```diff
@@ -1,16 +1,110 @@
 package controllers
 
 import (
+	"errors"
+	"fmt"
+	"os/exec"
+	"path"
+
+	"github.com/devclub-iitd/DeployBot/src/discord"
+	"github.com/devclub-iitd/DeployBot/src/helper"
 	"github.com/devclub-iitd/DeployBot/src/history"
+	"github.com/devclub-iitd/DeployBot/src/slack"
+	log "github.com/sirupsen/logrus"
 )
 
-func redeploy(callbackID string, data map[string]interface{}) {
-	stop(callbackID, data)
-	state, _ := history.GetState(data["git_repo"].(string))
+func redeploy(params *deployAction) {
+	state, _ := history.GetState(params.data["git_repo"].(string))
+
 	if state.Status == "stopped" {
-		data["subdomain"] = state.Subdomain
-		data["access"] = state.Access
-		data["server_name"] = state.Server
-		deploy(callbackID, data)
+		deploy(params)
+		return
+	}
+
+	params.data["subdomain"] = state.Subdomain
+	params.data["access"] = state.Access
+	params.data["server_name"] = state.Server
+
```

Comment:
I am thinking about a case where the state would have empty values for these field. That would probably happen if we call redeploy on a repo that hasn't been deployed. It's a corner case which I don't think would get triggered if we are careful about the usage, but in any case can you make sure this is handled here(and anywhere else where the same issue can happen)